sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_mgmt_async.py

```diff
@@ -0,0 +1,152 @@
+        return self
+
+    @classmethod
+    def _from_connection_string(cls, conn_str: str, **kwargs: Any # pylint: disable=docstring-keyword-should-match-keyword-only
+                                )-> "ServiceBusManagementOperationClient":
+        """Create a ServiceBusManagementOperation from a connection string.
+        """
+        constructor_args = cls._convert_connection_string_to_kwargs(conn_str, **kwargs)
+        return cls(**constructor_args)
+
+    def _create_handler(self, auth):
+        self._handler = self._amqp_transport.create_amqp_client_async(
+            config=self._config,
+            target=self._entity_uri,
+            auth=auth,
+            properties=self._properties,
+            retry_policy=self._error_policy,
+            client_name=self._name,
+        )
+
+    async def __aenter__(self) -> "ServiceBusManagementOperationClient":
+        if self._shutdown.is_set():
+            raise ValueError(
+                "The handler has already been shutdown. Please use ServiceBusClient to create a new instance."
+            )
+        await self._open_with_retry()
+        return self
+
+    async def _open(self) -> None:
+        # pylint: disable=protected-access
+        if self._running:
+            return
+        if self._handler and not self._handler._shutdown:
+            await self._handler.close_async()
+        auth = None if self._connection else (await create_authentication(self))
+        self._create_handler(auth)
+        try:
+            await self._handler.open_async(connection=self._connection)
+            while not await self._handler.client_ready_async():
+                await asyncio.sleep(0.05)
+            self._running = True
+        except:
+            await self._close_handler()
+            raise
+
+        if self._auto_lock_renewer and self._session:
+            self._auto_lock_renewer.register(self, self.session)
+
+    async def _close_handler(self):
+        await super(ServiceBusManagementOperationClient, self)._close_handler()
+
+    async def close(self):
+        # type: () -> None
+        await super(ServiceBusManagementOperationClient, self).close()
+
+    async def get_sessions(
+        self,
+        *,
+        last_updated_time: datetime.datetime = datetime.datetime.max,
+        skip_num_sessions: int = 0,
+        max_num_sessions: int = 1,
+        timeout: Optional[float] = None,
+        **kwargs
+    ):
+        """Get sessions from the Service Bus entity.
+        :keyword last_updated_time: Filter sessions based on the last updated time.
+        :paramtype last_updated_time: datetime.datetime
+        :keyword skip_num_sessions: Skip this number of sessions.
+        :paramtype skip_num_sessions: int
+        :keyword max_num_sessions: Maximum number of sessions to return.
+        :paramtype max_num_sessions: int
+        :return: List of session ids.
+        :rtype: List[str]
+        """
+        if kwargs:
```

Comment:
only works on queues

Reply:
b'A sessionful message receiver cannot be created on an entity that does not require sessions. Ensure RequiresSession is set to true when creating a Queue or Subscription to enable sessionful behavior.`


the service error is great along with good documentation